app/src/main/java/bacon/john/bitcoinpricewatch/bitcoin/BitcoinCurrencyManager.java

```diff
@@ -0,0 +1,120 @@
+package bacon.john.bitcoinpricewatch.bitcoin;
+
+import org.json.JSONObject;
+
+import java.util.*;
+import org.json.*;
+import java.text.*;
+import io.realm.*;
+
+import bacon.john.bitcoinpricewatch.localstorage.LocalStorageService;
+
+/**
+ * @class BitcoinCurrencyManager
+ *
+ * Retrieves information about bitcoin in various currencies
+ */
+public class BitcoinCurrencyManager {
+    public interface CurrencyListCallback {
+        public void callback(List<BitcoinCurrencyModel> currencies, Exception error);
+    }
+    public interface CurrencyCallback {
+        public void callback(BitcoinCurrencyModel currency, Exception error);
+    }
+
+    private static BitcoinCurrencyManager sharedInstance;
+
+    public static BitcoinCurrencyManager sharedInstance() {
+        if (sharedInstance == null) {
+            sharedInstance = new BitcoinCurrencyManager();
+        }
+        return sharedInstance;
+    }
+
+    public List<BitcoinCurrencyModel> getCachedCurrencies() {
+        return LocalStorageService.sharedInstance().getRealm().where(BitcoinCurrencyModel.class).findAll();
+    }
+
+    public BitcoinCurrencyModel getCachedCurrency(String currencyCode) {
+        return LocalStorageService.sharedInstance().getRealm().
```

Comment:
Realm rocks! I opted for Realm vs User Prefs since it would be more scalable / flexible and almost as easy to interact with.